lib/srv/sess.go

```diff
@@ -1272,34 +1592,36 @@ func (s *session) getParties() (parties []*party) {
 	return parties
 }
 
-func newMultiWriter() *multiWriter {
-	return &multiWriter{writers: make(map[string]writerWrapper)}
+func NewMultiWriter() *MultiWriter {
+	return &MultiWriter{writers: make(map[string]writerWrapper)}
 }
 
-type multiWriter struct {
+type MultiWriter struct {
 	mu           sync.RWMutex
 	writers      map[string]writerWrapper
 	recentWrites [][]byte
+	OnError      func(string, error)
 }
 
 type writerWrapper struct {
 	io.WriteCloser
 	closeOnError bool
+	id           string
 }
 
-func (m *multiWriter) addWriter(id string, w io.WriteCloser, closeOnError bool) {
+func (m *MultiWriter) AddWriter(id string, w io.WriteCloser, closeOnError bool) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.writers[id] = writerWrapper{WriteCloser: w, closeOnError: closeOnError}
+	m.writers[id] = writerWrapper{WriteCloser: w, closeOnError: closeOnError, id: id}
 }
 
-func (m *multiWriter) deleteWriter(id string) {
+func (m *MultiWriter) DeleteWriter(id string) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	delete(m.writers, id)
```

Comment:
Since the writer is a `WriteCloser`, should you close it before you stop tracking it, or is that up to a different part of the code?

Reply:
Additionally, should `MultiWriter` itself have a `Close() error` func?